src/main/java/org/gaul/s3proxy/requesthandlers/S3BucketListHandler.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright 2014-2017 Andrew Gaul <andrew@gaul.org>
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gaul.s3proxy.requesthandlers;
+
+import java.io.IOException;
+import java.io.Writer;
+import java.util.Date;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import javax.xml.stream.XMLStreamException;
+import javax.xml.stream.XMLStreamWriter;
+
+import org.gaul.s3proxy.S3Exception;
+import org.gaul.s3proxy.XMLUtils;
+import org.jclouds.blobstore.BlobStore;
+import org.jclouds.blobstore.domain.PageSet;
+import org.jclouds.blobstore.domain.StorageMetadata;
+
+public final class S3BucketListHandler extends AbstractS3BucketHandler {

```

Comment:
1. Always using objects are preferred over static utility methods in an object-oriented model. In fact, Jclouds also uses objects for these operations.
2. If we categorize them as larger entities like bucket or object, it will again increase the complexity of the code if we keep on adding new bucket or object functionalities in future.